src/commands/tail/mod.rs

```diff
@@ -51,3 +51,79 @@ fn find_open_port(requested: Option<u16>, default: u16) -> Result<u16, failure::
         Ok(socket.local_addr()?.port())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    // These tests are extremely stateful; since what we are testing is how this function behaves
+    // when requested ports are unavailable, and since our tests run concurrently, each test uses
+    // unique ports to avoid collisions. There are two possible solutions to this problem:
+    // 1. require that these tests be run serially, and find a way to clean up bound ports
```

Comment:
cleaning up bound ports should happen automatically since i believe TcpListener::bind implements Drop

Reply:
it does; what i'm not certain of is how i can get these tests in particular to run serially and sandboxed from all other tests in the suite. nothing else currently binds ports, but y'know.

this is also just not a good unit testing practice.